### ISSUE TITLE
Incorrect size calculation in replay status messages

### ISSUE THREAD

**Issue by HarukaMa** [bitshares/bitshares-core#2020]
**Bug Description**
The "by size" calculation during replay seems incorrect; it sometimes would exceed 100%.

**Impacts**
Describe which portion(s) of BitShares Core may be impacted by this bug. Please tick at least one box.
- [ ] API (the application programming interface)
- [ ] Build (the build process or something prior to compiled code)
- [ ] CLI (the command line wallet)
- [ ] Deployment (the deployment process after building such as Docker, Travis, etc.)
- [ ] DEX (the Decentralized EXchange, market engine, etc.)
- [ ] P2P (the peer-to-peer network for transaction/block propagation)
- [ ] Performance (system or user efficiency, etc.)
- [ ] Protocol (the blockchain logic, consensus, validation, etc.)
- [ ] Security (the security of system or user data, etc.)
- [x] UX (the User Experience)
- [ ] Other (please add below)

**Steps To Reproduce**
Steps to reproduce the behavior (example outlined below):
1. Start partial replay e.g. when restarting witness_node
2. Observe the progress log messages
3. "by size" part could get over 100% after saving the database

**Expected Behavior**
Calculations should be correct and below 100%.

**Host Environment**
Please provide details about the host environment. Much of this information can be found running: `witness_node --version`. 
 - Host OS:  Arch Linux
 - Host Physical RAM: 24 GB
 - BitShares Version: eb335192c08d9053afe1456613154c18e050b9f3
 - OpenSSL Version: 1.1.1d
 - Boost Version: 1.69
 
**Additional Context (optional)**
```
1804123ms th_a       db_management.cpp:70          reindex              ] Replaying blocks, starting at 41403590...
1915255ms th_a       db_management.cpp:138         reindex              ]    [by size: 99.81140%   76839201971 of 76984398295]   [by num: 99.84606%   41410000 of 41473844]
2096498ms th_a       db_management.cpp:138         reindex              ]    [by size: 99.84391%   76864230460 of 76984398295]   [by num: 99.87017%   41420000 of 41473844]
2277343ms th_a       db_management.cpp:138         reindex              ]    [by size: 99.87679%   76889543876 of 76984398295]   [by num: 99.89429%   41430000 of 41473844]
2443995ms th_a       db_management.cpp:138         reindex              ]    [by size: 99.90739%   76913102038 of 76984398295]   [by num: 99.91840%   41440000 of 41473844]
2626439ms th_a       db_management.cpp:138         reindex              ]    [by size: 99.93922%   76937609728 of 76984398295]   [by num: 99.94251%   41450000 of 41473844]
2793716ms th_a       db_management.cpp:138         reindex              ]    [by size: 99.96893%   76960481179 of 76984398295]   [by num: 99.96662%   41460000 of 41473844]
2849808ms th_a       db_management.cpp:142         reindex              ] Writing database to disk at block 41463844
2890109ms th_a       db_management.cpp:144         reindex              ] Done
2975088ms th_a       db_management.cpp:138         reindex              ]    [by size: 100.01359%   76994860494 of 76984398295]   [by num: 99.99073%   41470000 of 41473844]
3020547ms th_a       db_management.cpp:159         reindex              ] Done reindexing, elapsed time: 1216.42450099999996382 sec
```
Notice that before saving the database, size was increasing steadily without any spikes, but during the last 10000 blocks the size increased significantly. The blockchain itself did not have any activity spike during that period.

## CORE TEAM TASK LIST
- [ ] Evaluate / Prioritize Bug Report
- [ ] Refine User Stories / Requirements
- [ ] Define Test Cases
- [ ] Design / Develop Solution
- [ ] Perform QA/Testing
- [ ] Update Documentation

**Comment by pmconrad** [comment]
Thanks.

The reason is that [this line](https://github.com/bitshares/bitshares-core/blob/3.3.2/libraries/chain/db_management.cpp#L90) assumes that the file position is after the end of the previous block in the blocks database. However, after `undo_point` we switch from `apply_block` to `push_block` which also re-appends the blocks to the block database and thus moves the file point beyond the original file size.

This wastes some disk space but it nothing to worry about.

**Comment by abitmore** [comment]
> ... after `undo_point` we switch from `apply_block` to `push_block` which also re-appends the blocks to the block database and thus moves the file point beyond the original file size.
>
>This wastes some disk space but it nothing to worry about.

I think it's best to not re-append the blocks to the block database when replaying. It's a minor issue though. Low priority imo.

**Comment by abitmore** [comment]
Fixed by #2246.
